iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java

```diff
@@ -32,14 +35,128 @@ public class PublicNetworkAccessCheck extends AbstractArmResourceCheck {
   private static final Set<String> SENSITIVE_VALUES = Set.of("Enabled", "EnabledForSessionHostsOnly", "EnabledForClientsOnly");
 
   @Override
+  // Method has 101 lines, which is greater than 100 authorized - it will be refactored when SONARIAC-890 will be done
+  @SuppressWarnings("java:S138")
   protected void registerResourceConsumer() {
-    register("Microsoft.DesktopVirtualization/hostPools",
-      (ctx, resource) -> PropertyUtils.value(resource, "publicNetworkAccess")
-        .filter(PublicNetworkAccessCheck::isSensitiveStringLiteral)
-        .ifPresent(value -> ctx.reportIssue(value, "Make sure allowing public network access is safe here.")));
+    register("Microsoft.DesktopVirtualization/hostPools", checkPublicNetworkAccess());
+
+    register("Microsoft.AgFoodPlatform/farmBeats", checkPublicNetworkAccessSimplified());
```

Comment:
A bit heavy to read :sweat_smile: Maybe put all those resources in a list and have a `.forEach()` call on it, WDYT?

Reply:
The new method `protected void register(List<String> resourceTypes, BiConsumer<CheckContext, ResourceDeclaration> consumer)` will be provided by Nils, and I will use it. But ok I can do it for now.